### DISCUSSION TITLE
Is it ok to send messages programmatically with `Command::perform(std::future::ready(1), Message::Number)`

### DISCUSSION THREAD

**Discussion by lucatrv** [hecrj/iced#1895] [Q&A] [answered]
Sometimes I feel the need to send some messages programmatically from within the `update` method, based on internal logic. Is it ok to use the example in subject? I read question #1172 but I wonder if it applies also to the example in subject, as I do not see how this could "break all guarentees that the Elm architechture provides".

**Comment by hecrj** [comment, marked as answer]
Why? Create a function for the logic that processes `Message::Number` and just call it when needed.

**Reply by hecrj** [reply]
Or maybe just duplicate the code! Copy and paste. Abstractions aren't always the right tool.

**Comment by lucatrv** [comment]
Thanks @hecrj for your hints, your answers helped me think the `Elm` / `Iced` way. I avoided code duplication by creating functions which encapsulate common logic elaborations, and also by matching common behaving messages with multiple patterns.
